repo2docker/buildpacks/base.py

```diff
@@ -349,7 +350,21 @@ def _filter_tar(tar):
             src_path = os.path.join(os.path.dirname(__file__), *src_parts)
             tar.add(src_path, src, filter=_filter_tar)
 
-        tar.add('.', 'src/', filter=_filter_tar)
+        _exclude_tar = None
+        if os.path.exists(".gitignore"):
+            gitignore_fh = open(".gitignore")
```

Comment:
can you use a context manager here so that the file gets closed?

Reply:
Typical docker builds use a separate `.dockerignore` and ignore `.gitignore`. Should we follow that pattern? I'd rather be as close to exact for what `docker build` would normally do as we can.

Reply:
@betatim : sure, I will push in another commit with a bit of refactoring, and also the tests.

@minrk : Thats more of a usability question. If `repo2docker` wants to be transparent to the user then, adding a separate `.dockerignore` might add to the confusion. You could add conditionals to check if both `.dockerignore` and `.gitignore` exist, and prioritize the `.dockerignore`, but if the repository just has a `.gitignore` then you use that instead.